### ISSUE TITLE
Markdown support is incomplete and inconsistent

### ISSUE THREAD

**Issue by jheer** [idyll-lang/idyll#34]
The use of Markdown syntax is a wonderful way to leverage people's existing familiarity. However, there are some gaps in coverage; examples included below. Some should be easy to address, others perhaps less so. At minimum, it seems important that the documentation eventually note any discrepancies, which seem likely to trip up new users.

### Missing Constructs

- Italic/bold styling using underscore (`_italic_`, `__bold__`).
- Unordered lists initiated by hyphen (`- List item`)
- Ordered lists (`1. List item`)
- Syntax highlighted code blocks (```` ```json ````)

### Inconsistent Constructs

- Using an asterisk as the first character of a line in Idyll results in an unordered list. Other Markdown flavors will produce italicized text if a line of text consists solely of `*italic*`, and bold text if a line consists solely of `**bold**`.
- Hyperlinks `[text](url)` collide with Idyll's component syntax. Trying to include a Markdown hyperlink can result in an Idyll compilation error, which could be jarring for some. Perhaps the use of parentheses and lack of closing bracket `/]` is syntactically unique here? If so, standard links might be supported without modification to component definitions.

**Comment by mathisonian** [comment]
Thanks @jheer - 

I have issues for most of these in the [compiler repo](https://github.com/idyll-lang/idyll-compiler), but I think its good to keep this issue open until we reach a reasonable parity or have the differences well documented. 

The biggest thing right now is restructuring how the asterisk characters are parsed to make it more consistent with other markdown implementations, as you note. This is being tracked in https://github.com/idyll-lang/idyll-compiler/issues/3, I think once that is done `Italic/bold styling using underscore` and `ordered lists`, and `Unordered lists initiated by hyphen` should follow fairly easily. 

I've just opened a PR addressing markdown style hyperlinks, and will open a separate issue for `Syntax highlighted code blocks`.

**Comment by mathisonian** [comment]
https://github.com/idyll-lang/idyll-compiler/pull/14 addresses most of these issues. The outstanding ones are:

* Syntax highlighting (tracked in https://github.com/idyll-lang/idyll/issues/38)
and
* Having clear documentation of differences from markdown (tracked in https://github.com/idyll-lang/idyll-lang.github.io/issues/1)

**Comment by rreusser** [comment]
Added bold *and* italic via idyll-lang/idyll-compiler#28
